### PR TITLE
Ewl 842  toolkit full bracket improve navigation

### PR DESCRIPTION
**Pull request by Xb0X**
Add the carrousel system to the round table component

# About the type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/components/Carousel/index.js [hunk 1]
```diff
@@ -54,13 +55,16 @@ const StyledSlidesTrack = styled.div`
...
-  display: inline-block;
+  display: ${props => (props.flex ? 'flex' : 'inline-block')};
   vertical-align: top;
   width: auto;
   margin-right: ${props => props.margin}px;
   &:last-of-type {
     margin-right: 0;
   }
+  &:first-of-type {
+    padding-left: ${props => (props.offsetLeft ? `${props.offsetLeft}px;` : '0px;')};
```
Comment (JulienKode) [inline src/components/Carousel/index.js]: What about having just a `0` ?

(Sorry my comment is may be too picky)

Comment (JulienKode) [inline src/components/Carousel/index.js]: Seems that we have 3 `;` ?

File: src/components/Carousel/index.js [hunk 2]
```diff
@@ -395,16 +435,23 @@ export default class Carousel extends React.Component {
...
             alignCenter={alignCenter}
             disableNavigation={disableNavigation}
+            flex={flex}
           >
             {children.map((child, index) => (
-              /* eslint-disable-next-line react/no-array-index-key */
-              <StyledSlide key={index} margin={slideMargin} data-test={`carousel-item-${index}`}>
+              <StyledSlide
+                /* eslint-disable-next-line react/no-array-index-key */
+                key={index}
```
Comment (JulienKode) [inline src/components/Carousel/index.js]: Any other way than having a key as `index` ? 🤔
I know that it was already here

File: src/components/Carousel/index.js (hunk 1 above)
Comment (leozero) [reply src/components/Carousel/index.js]: Done ;)

Comment (leozero) [reply src/components/Carousel/index.js]: Done ;)

File: src/components/Carousel/index.js [hunk 3]
```diff
@@ -377,11 +395,33 @@ export default class Carousel extends React.Component {
...
     const { left, trackWidth, disableNavigation, slides, isDragging } = this.state;
+    const defaultNavigationEnabled = !disableNavigation && !absoluteNavigation && withArrow;
+    const absoluteNavigationEnabled = !disableNavigation && absoluteNavigation && withArrow;
+
     return (
       <StyledWrapper className={className} slides={slides} data-test="carousel">
-        {!disableNavigation && withArrow && (
+        {absoluteNavigationEnabled && (
+          <StyledAbsoluteArrowWrapper>
+            <StyledArrowLeft onClick={() => this.slide(true)} data-test="carousel-arrow-left" absoluteNavigation>
```
[the remark this answers is not in the document]
Comment (leozero) [reply src/components/Carousel/index.js]: Done

File: src/components/Carousel/index.js [hunk 4]
```diff
@@ -377,11 +395,33 @@ export default class Carousel extends React.Component {
...
+      slideMargin,
+      className,
+      withArrow,
+      alignCenter,
+      flex,
+      absoluteNavigation,
+      offsetLeft,
+    } = this.props;
     const { left, trackWidth, disableNavigation, slides, isDragging } = this.state;
+    const defaultNavigationEnabled = !disableNavigation && !absoluteNavigation && withArrow;
```
[the remark this answers is not in the document]
Comment (leozero) [reply src/components/Carousel/index.js]: Probably later ;)

File: src/components/Carousel/index.js (hunk 2 above)
Comment (leozero) [reply src/components/Carousel/index.js]: We have no id or things like this, but we'll try to find a solution later ;-)

**Pull request merged**